CMake/resolve_dependency_modules/qatzstd.cmake

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+include_guard(GLOBAL)
+
+include(Findzstd)
+
+set(VELOX_QATZSTD_BUILD_VERSION 0.0.1)
+set(VELOX_QATZSTD_BUILD_SHA256_CHECKSUM
+    c15aa561d5139d49f896315cbd38d02a51b636e448abfa58e4d3a011fe0424f2)
+string(CONCAT VELOX_QATZSTD_SOURCE_URL
+              "https://github.com/intel/QAT-ZSTD-Plugin/archive/refs/tags/"
+              "v${VELOX_QATZSTD_BUILD_VERSION}.tar.gz")
+
+resolve_dependency_url(QATZSTD)
+
+message(STATUS "Building QATZSTD from source")
+
+ProcessorCount(NUM_JOBS)
+set_with_default(NUM_JOBS NUM_THREADS ${NUM_JOBS})
+find_program(MAKE_PROGRAM make REQUIRED)
+
+set(QATZSTD_SOURCE_DIR "${CMAKE_CURRENT_BINARY_DIR}/_deps/qatzstd-src")
+set(QATZSTD_INCLUDE_DIR "${QATZSTD_SOURCE_DIR}/src")
+set(QATZSTD_STATIC_LIB_TARGETS "${QATZSTD_SOURCE_DIR}/src/libqatseqprod.a")
+set(QATZSTD_MAKE_ARGS "ENABLE_USDM_DRV=1" "ZSTDLIB=${ZSTD_INCLUDE_DIR}")
+
+ExternalProject_Add(
+  qatzstd
+  URL ${VELOX_QATZSTD_SOURCE_URL}
+  URL_HASH ${VELOX_QATZSTD_BUILD_SHA256_CHECKSUM}
+  SOURCE_DIR ${QATZSTD_SOURCE_DIR}
+  BINARY_DIR ${QATZSTD_BINARY_DIR}
+  CONFIGURE_COMMAND ""
+  BUILD_COMMAND ${MAKE_PROGRAM} ${QATZSTD_MAKE_ARGS}
+  INSTALL_COMMAND ""
+  BUILD_BYPRODUCTS ${QATZSTD_STATIC_LIB_TARGETS}
+  BUILD_IN_SOURCE 1)
+
+add_library(qatzstd::qatzstd STATIC IMPORTED)
+
+# The include directory must exist before it is referenced by a target.
+file(MAKE_DIRECTORY "${QATZSTD_INCLUDE_DIR}")
+
+find_library(
+  USDM_DRV_LIBRARY REQUIRED
+  NAMES usdm_drv_s
+  PATHS "$ENV{ICP_ROOT}/build"
+  NO_DEFAULT_PATH)
+find_library(
+  QAT_S_LIBRARY REQUIRED
+  NAMES qat_s
+  PATHS "$ENV{ICP_ROOT}/build"
+  NO_DEFAULT_PATH)
```

Comment:
Could you explain this bit? What are these libraries and where do they come from? Could we use `find_package`?

Reply:
I assume these missing deps are leading to this build error? https://app.circleci.com/pipelines/github/facebookincubator/velox/30869/workflows/1e47f8dc-9eb5-42c3-be30-1fac458d6e33/jobs/194704?invite=true#step-105-553

Reply:
> Could you explain this bit? What are these libraries and where do they come from? Could we use `find_package`?

these are packages from QAT driver. ICP_ROOT is the path to QAT driver. This PR is only suppose to build on QAT enabled machines for evaluation and benchmarking. You are right about the CICD error. It's due to not able to find the packages on a non-qat machine. 

@yyang52, feel free to add if i miss anything.

Reply:
Alright, makes sense for now, before this is merged this should also use resolve_dependency/find_package and use targets.